x/cdp/keeper/fees.go

```diff
@@ -27,6 +28,46 @@ func (k Keeper) CalculateFees(ctx sdk.Context, principal sdk.Coins, periods sdk.
 	return newFees
 }
 
+// TODO UpdateFeesForRiskyCdps
+// Select cdps with 10% of the liquidation ratio (ie. 150% * 0.1 = 15% = 165% or lower)
+// call calculate fees for each of those cdps
+// store the result of calculate fees in cdp.AccumulatedFees
+// set cdp.FeesLastUpdated to the current block time (ie. ctx.BlockTime())
+
+// UpdateFeesForRiskyCdps calculates fees for risky CDPs
+// The overall logic is first select the CDPs with 10% of the liquidation ratio
+// Then we call calculate fees on each of those CDPs
+// Next we store the result of the fees in the cdp.AccumulatedFees field
+// Finally we set the cdp.FeesUpdated time to the current block time (ctx.BlockTime()) since that
+// is when we made the update
+// TODO - QUESTION - this method signature should only take (ctx sdk.Context, cp types.Params) as parameters? Is this correct?
+// TODO - QUESTION - is types.CollateralParam the correct type?
+func (k Keeper) UpdateFeesForRiskyCdps(ctx sdk.Context, cp types.CollateralParam) {
+
+	// first calculate the target ratio based on liquidation ratio plus ten percent
+	value, err := sdk.NewDecFromStr("1.1")
+	if err != nil {
+		fmt.Printf("got error: %s", err) //  TODO - QUESTION - is there another method we should use for error logging instead of
+		// just using printf?
+	}
+	targetRatio := k.getLiquidationRatio(ctx, cp.Denom).Mul(value) // corresponds to 110% of the liquidation ratio
+
+	// now iterate over all the cdps based on collateral ratio
+	k.IterateCdpsByCollateralRatio(ctx, cp.Denom, targetRatio, func(cdp types.CDP) bool {
+		// get the number of periods
+		periods := sdk.NewInt(ctx.BlockTime().Unix()).Sub(sdk.NewInt(cdp.FeesUpdated.Unix()))
+		// now calcuate and store additional fees
+		additionalFees := k.CalculateFees(ctx, cdp.Principal, periods, cp.Denom)
+		// now add the additional fees to the accumulated fees for the cdp
+		cdp.AccumulatedFees.Add(additionalFees)
+		// and set the fees updated time to the current block time since we just updated it
+		cdp.FeesUpdated = ctx.BlockTime()
```

Comment:
```suggestion
		cdp.FeesUpdated = ctx.BlockTime()
		collateralToDebtRatio := k.CalculateCollateralToDebtRatio(ctx, cdp.Collateral, cdp.Principal.Add(cdp.AccumulatedFees))
	        k.SetCdpAndCollateralRatioIndex(ctx, cdp, collateralToDebtRatio)
```

Note, you haven't updated the state of the CDP in the store. https://github.com/Kava-Labs/kava/blob/4d21230925baf19d55d0fe3ed59703939348dfb6/x/cdp/keeper/deposit.go#L47 is a pretty generic example of updating the cdp state after altering the fees.